app/services/maat_link_creator.rb

```diff
@@ -1,13 +1,20 @@
 # frozen_string_literal: true
 
 class MaatLinkCreator < ApplicationService
-  def initialize(laa_reference_id:)
-    @laa_reference = LaaReference.find(laa_reference_id)
+  attr_reader :laa_reference
+
+  def initialize(defendant_id, user_name, maat_reference)
+    @laa_reference = LaaReference.new(
+      defendant_id: defendant_id,
+      user_name: user_name,
+      maat_reference: maat_reference.presence || LaaReference.generate_linking_dummy_maat_reference,
+    )
   end
 
   def call
     publish_laa_reference_to_queue unless laa_reference.dummy_maat_reference?
     post_laa_references_to_common_platform
+    persist_laa_reference!
```

Comment:
So if `post_laa_references_to_common_platform` fails, do we definitely not hit this line?

If response from RecordLaaReference.call is status 500, is the default behaviour to throw an error?